app/controllers/items_controller.rb

```diff
@@ -47,14 +47,24 @@ def embargo_form
   end
 
   def set_collection
-    @object.collections.each { |collection| @object.remove_collection(collection.pid) } # first remove any existing collections
-    if params[:collection].present?
-      @object.add_collection(params[:collection]) # collection provided, so add it
-      response_message = 'Collection successfully set.'
-    else
-      response_message = 'Collection(s) successfully removed.' # no collection selected from drop-down, so don't bother adding a new one
-    end
-    save_and_reindex
+    object_client = Dor::Services::Client.object(@object.pid)
```

Comment:
There's a lot going on here that may be ripe for eventual extraction but I don't think that's necessary at this point.